src/engine/cancun.md

```diff
@@ -0,0 +1,133 @@
+# Engine API -- Cancun
+
+Engine API changes introduced in Cancun.
+
+This specificaiton is based on and extends [Engine API - Shanghai](./shanghai.md) specification.
+
+## Table of contents
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Structures](#structures)
+  - [ExecutionPayloadV3](#executionpayloadv3)
+  - [BlobsBundleV1](#blobsbundlev1)
+- [Methods](#methods)
+  - [engine_newPayloadV3](#engine_newpayloadv3)
+    - [Request](#request)
+    - [Response](#response)
+    - [Specification](#specification)
+  - [engine_getPayloadV3](#engine_getpayloadv3)
+    - [Request](#request-1)
+    - [Response](#response-1)
+    - [Specification](#specification-1)
+  - [Deprecate `engine_exchangeTransitionConfigurationV1`](#deprecate-engine_exchangetransitionconfigurationv1)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Structures
+
+### ExecutionPayloadV3
+
+This structure has the syntax of [`ExecutionPayloadV2`](./shanghai.md#executionpayloadv2) and append two new fields: `dataGasUsed` and `excessDataGas`.
+
+- `parentHash`: `DATA`, 32 Bytes
+- `feeRecipient`:  `DATA`, 20 Bytes
+- `stateRoot`: `DATA`, 32 Bytes
+- `receiptsRoot`: `DATA`, 32 Bytes
+- `logsBloom`: `DATA`, 256 Bytes
+- `prevRandao`: `DATA`, 32 Bytes
+- `blockNumber`: `QUANTITY`, 64 Bits
+- `gasLimit`: `QUANTITY`, 64 Bits
+- `gasUsed`: `QUANTITY`, 64 Bits
+- `timestamp`: `QUANTITY`, 64 Bits
+- `extraData`: `DATA`, 0 to 32 Bytes
+- `baseFeePerGas`: `QUANTITY`, 256 Bits
+- `blockHash`: `DATA`, 32 Bytes
+- `transactions`: `Array of DATA` - Array of transaction objects, each object is a byte list (`DATA`) representing `TransactionType || TransactionPayload` or `LegacyTransaction` as defined in [EIP-2718](https://eips.ethereum.org/EIPS/eip-2718)
+- `withdrawals`: `Array of WithdrawalV1` - Array of withdrawals, each object is an `OBJECT` containing the fields of a `WithdrawalV1` structure.
+- `dataGasUsed`: `QUANTITY`, 64 bits
+- `excessDataGas`: `QUANTITY`, 64 Bits
+- `parentBeaconBlockRoot`: `DATA`, 32 Bytes
```

Comment:
I'm having trouble checking out this PR locally, so I can follow up with the following changes:

* updating to `PayloadAttributesV3` with the `parent_beacon_block_root`

Reply:
having to make changes in another PR #423

Reply:
ignore the above -- lets just move this from here to down below, reflecting my two outstanding suggestions on the PR currently